test/e2e_node/restart_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"k8s.io/kubernetes/test/e2e/framework"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"k8s.io/kubernetes/pkg/api"
+	"os/exec"
+)
+
+var _ = framework.KubeDescribe("Restart [Serial] [Slow] [Disruptive]", func() {
+	const (
+		POD_COUNT             = 100
+		POD_CREATION_INTERVAL = 100 * time.Millisecond
+		RECOVER_TIMEOUT       = 3 * time.Minute
+	)
+
+	f := framework.NewDefaultFramework("restart-test")
+	Context("Docker Daemon", func() {
+		Context("Network", func() {
+			It("should recover from ip leak", func() {
+				By("Creating batch pods on node")
+				pods := newTestPods(POD_COUNT, framework.GetPauseImageNameForHostArch(), "restart-docker-test")
+				createBatchPodWithRateControl(f, pods, POD_CREATION_INTERVAL)
+				defer deletePodsSync(f, pods)
+
+				By("Restart docker daemon 3 times and verify pods")
+				for i := 0; i < 3; i += 1 {
+					By("Restarting Docker Daemon")
+					// TODO: Find a uniform way to deal with systemctl/initctl/service operations. #34494
+					if stdout, err := exec.Command("sudo", "service", "docker", "restart").CombinedOutput(); err != nil {
```

Comment:
Well. Not really. Node-e2e-test assumes the OS has docker. Hence it did not put any docker checker in place. I just tried to pkill docker on the node e2e box, it could not come back.